### PR TITLE
FromStr implementation example

### PR DESCRIPTION
**Pull request by donniebishop**
Referencing #29375. Added example implementation of FromStr trait to API Documentation

### CODE REVIEW

File: src/libcore/str/mod.rs [hunk 1]
```diff
@@ -35,6 +35,39 @@ pub mod pattern;
...
+///
+/// impl FromStr for Point {
+///     type Err = ParseIntError;
+///
+///     fn from_str(s: &str) -> Result<Self, Self::Err> {
+///         let coords: Vec<&str> = s.trim_matches(|p| p == '(' || p == ')' )
+///                                  .split(",")
+///                                  .collect();
+///
+///         let x_fromstr = try!(coords[0].parse::<i32>());
```
Comment (steveklabnik) [inline src/libcore/str/mod.rs]: `?` should be used instead of `try!`.

File: src/libcore/str/mod.rs [hunk 2]
```diff
@@ -35,6 +35,39 @@ pub mod pattern;
...
+///         let y_fromstr = try!(coords[1].parse::<i32>());
```
Comment (steveklabnik) [inline src/libcore/str/mod.rs]: same here

**Review by steveklabnik — CHANGES_REQUESTED** [review]
Looks great! Just a tiny change. Thanks so much!

**Comment by donniebishop** [comment]
Funny enough, this is how I originally mocked it up, but opted for the more verbose option on the side of caution haha

Will make a quick change and recommit

Thanks for the feedback!

**Comment by steveklabnik** [comment]
@bors: r+ rollup

ha! awesome, thanks a ton!

**Pull request merged**